src/status_im/ui/screens/activity_center/notification/contact_verification/view.cljs

```diff
@@ -17,24 +15,16 @@
 
 (defn- context-tags
   [challenger? {:keys [author contact-verification-status]}]
-  (let [contact (rf/sub [:contacts/contact-by-identity author])]
-    [[quo2/user-avatar-tag
-      {:color          :purple
-       :override-theme :dark
-       :size           :small
-       :style          style/user-avatar-tag
-       :text-style     style/user-avatar-tag-text}
-      (activity-center.utils/contact-name contact)
-      (multiaccounts/displayed-photo contact)]
-     (if challenger?
-       (when (or (= contact-verification-status constants/contact-verification-status-accepted)
-                 (= contact-verification-status constants/contact-verification-status-trusted)
-                 (= contact-verification-status constants/contact-verification-status-untrustworthy))
-         (str (str/lower-case (i18n/label :t/replied)) ":"))
-       (when (or (= contact-verification-status constants/contact-verification-status-accepted)
-                 (= contact-verification-status constants/contact-verification-status-pending)
-                 (= contact-verification-status constants/contact-verification-status-declined))
-         (str (i18n/label :t/identity-verification-request-sent) ":")))]))
+  [[common/user-avatar-tag author]
+   (if challenger?
+     (when (or (= contact-verification-status constants/contact-verification-status-accepted)
+               (= contact-verification-status constants/contact-verification-status-trusted)
+               (= contact-verification-status constants/contact-verification-status-untrustworthy))
+       (str (str/lower-case (i18n/label :t/replied)) ":"))
+     (when (or (= contact-verification-status constants/contact-verification-status-accepted)
+               (= contact-verification-status constants/contact-verification-status-pending)
+               (= contact-verification-status constants/contact-verification-status-declined))
+       (str (i18n/label :t/identity-verification-request-sent) ":")))])
```

Comment:
Can we calculate the `when` in a `let` instead of repeating it?

Reply:
The checks are made against different constants. Where do you see duplication @ibrkhalil?

Reply:
My bad, I missed that